features/ui/select/select.tsx

```diff
@@ -0,0 +1,256 @@
+    width="12"
+    height="8"
+    viewBox="0 0 12 8"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M1 1.5L6 6.5L11 1.5"
+      stroke="#667085"
+      strokeWidth="1.66667"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    />
+  </DownCaret>
+);
+
+const Placeholder = ({ children, ...props }: PlaceholderProps) => (
+  <components.Placeholder {...props}>
+    {props.selectProps.iconSrc && (
+      <Icon src={props.selectProps.iconSrc} alt="" />
+    )}
+    {children}
+  </components.Placeholder>
+);
+
+const customStyles: StylesConfig = {
+  control: (provided, state) => ({
+    ...provided,
+    maxWidth: "20rem",
+    borderWidth: "1px",
+    borderColor: state.selectProps.error
+      ? `${color("error", 300)({ theme })}`
+      : state.isFocused
+      ? `${color("primary", 300)({ theme })}`
+      : `${color("gray", 300)({ theme })}`,
+    borderRadius: "8px",
+    boxShadow: state.isFocused
+      ? state.selectProps.error
+        ? "0px 1px 2px rgba(16, 24, 40, 0.05), 0px 0px 0px 4px #FEE4E2"
+        : "0px 1px 2px rgba(16, 24, 40, 0.05), 0px 0px 0px 4px #F4EBFF;"
+      : "0px 1px 2px rgba(16, 24, 40, 0.05)",
+    padding: "0.625rem 0.8125rem",
+    margin: "6px 0",
+    backgroundColor: state.isDisabled
+      ? `${color("gray", 50)({ theme })}`
+      : "#FFFFFF",
+
+    "&:hover": {
+      borderColor: state.selectProps.error
+        ? `${color("error", 300)({ theme })}`
+        : state.isFocused
+        ? `${color("primary", 300)({ theme })}`
+        : `${color("gray", 300)({ theme })}`,
+      cursor: "pointer",
+    },
+  }),
+
+  menu: (provided) => ({
+    ...provided,
+    maxWidth: "20rem",
+    boxShadow:
+      "0px 12px 16px -4px rgba(16, 24, 40, 0.1), 0px 4px 6px -2px rgba(16, 24, 40, 0.05)",
+    borderRadius: "8px",
+    margin: "0.5rem 0",
+  }),
+
+  menuList: (provided) => ({
+    ...provided,
+    maxHeight: "20rem",
+  }),
+
+  option: (provided, state) => ({
+    ...provided,
+    color: `${color("gray", 900)({ theme })}`,
+    lineHeight: "1.5rem",
+    padding: "0.625rem 0.875rem",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "space-between",
+    backgroundColor:
+      state.isFocused || state.isSelected
+        ? `${color("primary", 25)({ theme })}`
+        : "#FFFFFF",
+
+    "&:hover": {
+      backgroundColor: `${color("primary", 25)({ theme })}`,
+    },
+  }),
+
+  valueContainer: (provided) => ({
+    ...provided,
+    margin: "0",
+    padding: "0",
+  }),
+
+  placeholder: (provided) => ({
+    ...provided,
+    color: `${color("gray", 500)({ theme })}`,
+    lineHeight: "1.5rem",
+    display: "flex",
+    alignContent: "center",
+    margin: "0",
+  }),
+
+  singleValue: (provided) => ({
+    ...provided,
+    lineHeight: "1.5rem",
+    display: "flex",
+    alignContent: "center",
+    margin: "0",
+  }),
+};
+
+export function SelectComponent({
+  label,
+  hintMsg,
+  errorMsg,
+  error,
```

Comment:
Nitpick: This name is a bit misleading. I'd expect some kind of error object here. You could use something like `hasError` instead